src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py

```diff
@@ -74,12 +74,14 @@ def __init__(
         """
         self._name = name
         self._steps = steps
-        self._metric_fetchers = metric_fetchers
+        self._metric_fetchers: Dict[
+            str, MetricFetcher[GenericQuantity]
+        ] = metric_fetchers
         self._first_run = True
-        self._output_type = output_type
+        self._output_type: type[GenericQuantity] = output_type
```

Comment:
Oh, interesting that you can use `type` as a *generic type*, maybe you should replace the other `Type` uses?

Reply:
yup,  found this out near the end.  There are a lot of uses of `typing.*` in these files,  will replace them all in a separate PR.